File: hackerearth/practice/CompleteTheSyllabus.cpp
#include <iostream>

using namespace std;

int main(){
    string week[] = {"MONDAY","TUESDAY","WEDNESDAY","THURSDAY","FRIDAY","SATURDAY","SUNDAY"};

    int testcases;
    cin >> testcases;

    int work = 0;
    for(int i =  0;i < testcases;i++){
        cin >> work;

        int topics_per_week[7];
        for(int j = 0;j < 7;j++){
            cin >> topics_per_week[j];
        }

        int store_day = 0;
        int ittr = 0;
        while(work != 0){
            work -= topics_per_week[ittr];
            store_day = ittr;
            if(ittr == 6){
                ittr = 0;
            }
            ittr++;
        }

        cout << week[store_day] << endl;
    }
    return 0;
}

File: hackerearth/CompleteTheSyllabus.py
import sys

t = int(sys.stdin.readline().strip())

days = ["MONDAY", "TUESDAY", "WEDNESDAY", "THURSDAY", "FRIDAY", "SATURDAY", "SUNDAY"]


for i in range(t):
    k = int(sys.stdin.readline().strip())
    A = [int(x) for x in sys.stdin.readline().strip().split()]
    p = sum(A)

    if(k%p == 0):
        ans = 6
        while(A[ans] == 0):
            ans -= 1
        sys.stdout.write("%s\n"%(days[ans]))
        continue
    k = k%p
    kk = 0
    while True:
        k = k-A[kk]
        if(k<=0):
            sys.stdout.write("%s\n"%(days[kk]))
            break
        kk+=1